src/main/java/org/olf/folio/order/util/LookupUtil.java

```diff
@@ -6,110 +6,207 @@
+    }
+    
+    /**
+     * @param token
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws Exception
+     */
+    public HashMap<String, String> getReferenceValues(String token)
+            throws IOException, InterruptedException, Exception {
+
+        Map<String, String> table = new HashMap<String, String>();
+
+        Iterator<String> refTablesIterator = this.referenceTables.iterator();
+        while (refTablesIterator.hasNext()) {
+
+            String endpoint = refTablesIterator.next();
+            logger.trace("calling " + endpoint);
+            String response = getResponse(endpoint, token);
+
+            JSONObject jsonObject = new JSONObject(response);
+            // System.out.println(jsonObject.toString(3));
+
+            // TODO - IMPROVE THIS
+            // (0) IS THE NUMBER OF ITEMS FOUND
+            // (1) IS THE ARRAY OF ITEMS
+            // NOT A GOOD APPROACH
+            String elementName = (String) jsonObject.names().get(1);
+            JSONArray elements = jsonObject.getJSONArray(elementName);
+            Iterator elementsIterator = elements.iterator();
+            while (elementsIterator.hasNext()) {
+                JSONObject element = (JSONObject) elementsIterator.next();
+                String id = element.getString("id");
+                String name = element.getString("name");
+                if (endpoint.contains("locations")) {
+                    String code = element.getString("code");
+                    name = code + "-location";
+                }
+                // SAVING ALL OF THE 'NAMES' SO THE UUIDs CAN BE LOOKED UP
+
+                // logger.info("lookupTable name: "+ name);
+                table.put(name, id);
+            }
+        }
+        logger.trace("finished loading lookup table");
+
+        return (HashMap<String, String>) table;
+    }
+    
+    /**
+     * @param endpoint
+     * @param token
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws Exception
+     */
+    public HashMap<String,String> getBillingAddresses(String endpoint, String token) throws IOException, InterruptedException, Exception  { 
+        Map<String, String> table = new HashMap<String, String>();        
+        String response = this.apiService.callApiGet(endpoint, token); 
+         
+        JSONObject jsonObject = new JSONObject(response);
+         
+        
+        JSONArray configsArray = jsonObject.getJSONArray("configs");
+        Iterator elementsIterator = configsArray.iterator();
+        while (elementsIterator.hasNext()) {
+            JSONObject element = (JSONObject) elementsIterator.next();
+             
+            String value = element.getString("value");
+            String uuid = element.getString("id");
+            String configName = element.getString("configName");
+            if (StringUtils.equals(configName, "tenant.addresses")) {
```

Comment:
If we add the the `query` URL param suggested above, that would ensure only config entries of type `tenant.addresses` would be returned in the API response and then we can eliminate this test.

Reply:
I'll push a change that uses the query parameter.  *all* of the other query endpoints are using a limit though. I've thought about refactoring the LookupUtil class to that it's more explicit in it's queries and how it implements the lookkup table, but that'll take a bit of work for minimal gain.